pkg/planner/core/point_get_plan.go

```diff
@@ -2213,19 +2352,46 @@ func tryDeletePointPlan(ctx base.PlanContext, delStmt *ast.DeleteStmt, resolveCt
 		if ctx.GetSessionVars().TxnCtx.IsPessimistic {
 			pointGet.Lock, pointGet.LockWaitTime = getLockWaitTime(ctx, &ast.SelectLockInfo{LockType: ast.SelectLockForUpdate})
 		}
-		return buildPointDeletePlan(ctx, pointGet, pointGet.dbName, pointGet.TblInfo, delStmt.IgnoreErr)
-	}
-	if batchPointGet := tryWhereIn2BatchPointGet(ctx, selStmt, resolveCtx); batchPointGet != nil {
+		pointPlan, dbName, tbl, colsInWhereClause = pointGet, pointGet.dbName, pointGet.TblInfo, pointGet.colsInWhereClause
+	} else if batchPointGet := tryWhereIn2BatchPointGet(ctx, selStmt, resolveCtx); batchPointGet != nil {
 		if ctx.GetSessionVars().TxnCtx.IsPessimistic {
 			batchPointGet.Lock, batchPointGet.LockWaitTime = getLockWaitTime(ctx, &ast.SelectLockInfo{LockType: ast.SelectLockForUpdate})
 		}
-		return buildPointDeletePlan(ctx, batchPointGet, batchPointGet.dbName, batchPointGet.TblInfo, delStmt.IgnoreErr)
+		pointPlan, dbName, tbl, colsInWhereClause = batchPointGet, batchPointGet.dbName, batchPointGet.TblInfo, batchPointGet.colsInWhereClause
 	}
+	if pointPlan != nil {
+		// We don't check the output names, since they are expended to all columns of the table.
+		visitInfos := getVisitInfo(userName, hostName, dbName, tbl.Name.L, nil, nil, colsInWhereClause)
+		failpoint.Inject("point-get-visit-info", func(val failpoint.Value) {
+			if val.(bool) {
+				pointGetVisitInfo4Test = visitInfos
+			}
+		})
+		if checkFastPlanPrivilege(ctx, visitInfos...) != nil {
+			return nil
+		}
+		return buildPointDeletePlan(ctx, pointPlan, dbName, tbl, delStmt.IgnoreErr)
+	}
+
 	return nil
 }
 
 func buildPointDeletePlan(ctx base.PlanContext, pointPlan base.PhysicalPlan, dbName string, tbl *model.TableInfo, ignoreErr bool) base.Plan {
-	if checkFastPlanPrivilege(ctx, dbName, tbl.Name.L, mysql.SelectPriv, mysql.DeletePriv) != nil {
+	// There is no column-level DELETE privilege.
+	userName, hostName := auth.GetUserAndHostName(ctx.GetSessionVars().User)
+	visitInfos := []visitInfo{{
+		db:        dbName,
+		table:     tbl.Name.L,
+		column:    "",
```

Comment:
How about setting `column: ""` as a constant, with the constant name implying meaning like "no column-level privilege"?